Source/Table/DataSource/TableSectionsProvider.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  TableSectionsProvider.swift
+//  ReactiveDataDisplayManager
+//
+//  Created by Никита Коробейников on 12.02.2021.
+//  Copyright © 2021 Александр Кравченков. All rights reserved.
+//
+
+open class TableSectionsProvider: SectionsProvider {
+
+    public typealias GeneratorType = TableCellGenerator
+    public typealias HeaderGeneratorType = TableHeaderGenerator
+    public typealias FooterGeneratorType = TableFooterGenerator
+
+    open var sections: [Section<TableCellGenerator, TableHeaderGenerator, TableFooterGenerator>] = []
+
+    func addTableGenerators(with generators: [TableCellGenerator], choice section: СhoiceTableSection) {
+        switch section {
+        case .newSection(let header, let footer):
+            self.addGeneratorsInNewSection(generators: generators, for: header, footer: footer)
+        case .byIndex(let sectionIndex):
+            self.addGeneratorsInSection(by: sectionIndex, generators: generators)
+        case .lastSection:
+            self.sections.isEmpty || sections.count <= 0 ?
+            self.addTableGenerators(with: generators, choice: .newSection()) :
+            self.addGeneratorsInLastSection(generators: generators)
+        }
+    }
+
+    func addNewSection(section: Section<TableCellGenerator, TableHeaderGenerator, TableFooterGenerator>) {
```

Comment:
```suggestion
    private func addNewSection(section: Section<TableCellGenerator, TableHeaderGenerator, TableFooterGenerator>) {
```

Кажется этот метод и остальные что ниже должен стать приватным.
Иначе нет смысла вызывать добавление через choice или я ошибаюсь?

(приватность добавить предпочтительно через private extension)